spec/models/sale_price_spec.rb

```diff
@@ -35,4 +35,38 @@
     expect(money.money.currency).to eq(sale_price.currency)
   end
 
+  context 'touching associated product when destroyed' do
+    subject { -> { sale_price.destroy } }
+    let(:sale_price) { Timecop.travel(1.day.ago) { create(:sale_price) } }
+
+    it { is_expected.to change { sale_price.product.reload.updated_at } }
+
+    context 'when associated product has been destroyed' do
+      it 'does not touch product' do
+        product = sale_price.product
```

Comment:
@AlessioRocco no because I later use `product.reload`. If product or variant has been deleted (which is what the test does), `sale_price.product` will be nil and I cannot access the product anymore.